runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -656,18 +656,26 @@ private void resurrectCheckpoints() {
         try {
             for (Map.Entry<UUID, StreamMetaData> entry : streamsMetaData.entrySet()) {
                 cfs[i++] = CompletableFuture.runAsync(() -> {
-                    CheckPoint checkPoint = entry.getValue().getLatestCheckPoint();
-                    if (checkPoint == null) {
-                        log.info("resurrectCheckpoints[{}]: Truncated checkpoint for this stream",
-                                Utils.toReadableId(entry.getKey()));
-                        return;
-                    }
-
-                    // For now one by one read and apply
-                    for (long address : checkPoint.getAddresses()) {
-                        updateCorfuObject(getLogData(runtime, loadInCache, address));
-                    }
-                }, executorService);
+                    CheckPoint checkPoint = null;
+                    try {
+                        checkPoint = entry.getValue().getLatestCheckPoint();
+                        if (checkPoint == null) {
+                            log.info("resurrectCheckpoints[{}]: Truncated checkpoint for this stream",
+                                    Utils.toReadableId(entry.getKey()));
+                            return;
+                        }
+
+                        // For now one by one read and apply
+                        for (long address : checkPoint.getAddresses()) {
+                            updateCorfuObject(getLogData(runtime, loadInCache, address));
+                        }
+                    } catch (Throwable t) {
```

Comment:
Throwable is very broad, maybe we can be more specific?

Reply:
The thing here is that if anything goes wrong at this level (exception, error, whatever) we want to log it before this completes exceptionally. For instance we were seeing StackOverflowError which wouldn't have been captured if we caught any more specific exception.